python/cudf/cudf/core/frame.py

```diff
@@ -438,128 +438,241 @@ def _normalize_columns_and_scalars_type(self, other):
+            for column_name, other_column in zip(self.columns, other):
+                input_col = self._data[column_name]
+                if column_name in cond.columns:
+                    if is_categorical_dtype(input_col.dtype):
+                        if np.isscalar(other_column):
+                            try:
+                                other_column = input_col._encode(other_column)
+                            except ValueError:
+                                # When other is not present in categories.
+                                other_column = input_col.default_na_value()
+                        elif hasattr(other_column, "codes"):
+                            other_column = other_column.codes
+                        input_col = input_col.codes
+
+                    result = libcudf.copying.copy_if_else(
+                        input_col, other_column, cond._data[column_name]
+                    )
 
-                result = libcudf.copying.copy_if_else(
-                    input_col, otr_col, boolean_mask[cond_col_name]._column
-                )
+                    if is_categorical_dtype(self._data[column_name].dtype):
+                        result = build_categorical_column(
+                            categories=self._data[column_name].categories,
+                            codes=as_column(
+                                result.base_data, dtype=result.dtype
+                            ),
+                            mask=result.base_mask,
+                            size=result.size,
+                            offset=result.offset,
+                            ordered=self._data[column_name].ordered,
+                        )
+                else:
+                    from cudf._lib.null_mask import MaskState, create_null_mask
 
-                if is_categorical_dtype(self[in_col_name].dtype):
-                    result = build_categorical_column(
-                        categories=self[in_col_name]._column.categories,
-                        codes=as_column(result.base_data, dtype=result.dtype),
-                        mask=result.base_mask,
-                        size=result.size,
-                        offset=result.offset,
-                        ordered=self[in_col_name]._column.ordered,
+                    out_mask = create_null_mask(
+                        len(input_col), state=MaskState.ALL_NULL
                     )
+                    result = input_col.set_mask(out_mask)
+                out_df[column_name] = self[column_name].__class__(result)
 
-                out_df[in_col_name] = self[in_col_name].__class__(result)
-
-            return out_df
+            return self._mimic_inplace(out_df, inplace=inplace)
 
         else:
+            if isinstance(self, cudf.core.multiindex.MultiIndex):
+                raise NotImplementedError(
+                    ".where is not supported for MultiIndex operations"
+                )
```

Comment:
Could just have a `def where` in MultiIndex that raises a NotImplementedError

Reply:
Yep, felt the same initially. Added a dedicated `where` & `mask` for MultiIndex.